### DISCUSSION TITLE
ccls can't work when I generate the compile_commands.json at project top

### DISCUSSION THREAD

**Discussion by ycx-ycxiao** [neoclide/coc.nvim#4581] [Q&A] [answered]
My ccls server config is use the example which given in https://github.com/neoclide/coc.nvim/wiki/Language-servers
I can using "gd", "gr", "gi" success when I haven't generate the file compile_commands.json at project top.(Only some file can't find by ccls, this is why I generate the file). 
It's work fine after remove compile_commands.json and the follows ERROR for `jumpDefinition` will disappear.
Follow are server log, how can I fix this issue?
![image](https://user-images.githubusercontent.com/109899086/226817859-efb1f5f2-1697-487c-893a-a3862ff5e6c6.png)
![image](https://user-images.githubusercontent.com/109899086/226819850-a1551a81-fa71-48f0-b663-8850aab0d05b.png)

**Comment by fannheyward** [comment, marked as answer]
You shouldn't use coc-ccls, this extension is outdated and broken with coc.nvim.

**Reply by ycx-ycxiao** [reply]
Thanks for reply. Should I move to clangd? Or I should do some other things.

**Reply by fannheyward** [reply]
You can setup ccls in your coc-settings.json like https://github.com/neoclide/coc.nvim/wiki/Language-servers#ccobjective-c, or try clangd.

**Reply by ycx-ycxiao** [reply]
Sorry, maybe I need change some description for ccls. I'm not using plugin coc-ccls, I config it in my coc-settings.json. Above issue is happen on "setup ccls in your coc-settings.json". I don't know why it try to found `coc-ccls/package.json`  but the ERROR don't effect `jumpDefinition`.(Because I using  coc-settings.json to config ccls not plugin coc-ccls)

**Reply by ycx-ycxiao** [reply]
The first picture ERROR is because file "~/.config/coc/extensions/package.json" have coc-ccls  information, I detete the information then ERROR disappear.

**Reply by fannheyward** [reply]
For the second error, looks like ccls fails to response, check the trace logs https://github.com/neoclide/coc.nvim/wiki/Debug-language-server#using-output-channel

**Reply by ycx-ycxiao** [reply]
Yes, you are right. Ccls failed to response(the request from user failed).  I check the detail in log and fix this issue. Thank you very much.
![image](https://user-images.githubusercontent.com/109899086/227068190-8cb063c6-c247-4d37-b0a0-d67323c72078.png)
